torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,444 @@
+        with fiona.open(metadata_fn) as f:
+            for row in f:
+                fold = int(row["properties"]["Fold"])
+                if fold in self.folds:
+                    self.idxs.append(row["properties"]["ID_PATCH"])
+
+        files = []
+        for i in self.idxs:
+            suffix = f"{i}.npy"
+            files.append(
+                dict(
+                    s2=os.path.join(self.root, self.directory, self.prefix["s2"])
+                    + suffix,
+                    s1a=os.path.join(self.root, self.directory, self.prefix["s1a"])
+                    + suffix,
+                    s1d=os.path.join(self.root, self.directory, self.prefix["s1d"])
+                    + suffix,
+                    semantic=os.path.join(
+                        self.root, self.directory, self.prefix["semantic"]
+                    )
+                    + suffix,
+                    instance=os.path.join(
+                        self.root, self.directory, self.prefix["instance"]
+                    )
+                    + suffix,
+                )
+            )
+        return files
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the directory already exists
+        path = os.path.join(self.root, self.directory)
+        if os.path.exists(path):
+            return
+
+        # Check if zip file already exists (if so then extract)
+        filepath = os.path.join(self.root, self.filename)
+        if os.path.exists(filepath):
+            if self.checksum and not check_integrity(filepath, self.md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download and extract the dataset
+        self._download()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_url(
+            self.url,
+            self.root,
+            filename=self.filename,
+            md5=self.md5 if self.checksum else None,
+        )
+        extract_archive(os.path.join(self.root, self.filename), self.root)
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`NonGeoClassificationDataset.__getitem__`
```

Comment:
```suggestion
            sample: a sample returned by :meth:`__getitem__`
```
A class only needs to be specified if this class does not have a `__getitem__` method defined